apps/src/lab2/lab2Redux.ts

```diff
@@ -288,15 +288,25 @@ export const isReadOnlyWorkspace = (state: RootState) => {
   const isFrozen = !!state.lab.channel?.frozen;
   const isPredictLevel =
     state.lab.levelProperties?.predictSettings?.isPredictLevel || false;
+  let isReadonlyPredictLevel = isPredictLevel;
+  if (isPredictLevel) {
+    const isEditableAfterSubmit =
+      state.lab.levelProperties?.predictSettings?.editableAfterSubmit || false;
+    const hasSubmittedPredictResponse = state.predictLevel.hasSubmittedResponse;
+    // If the predict level is not editable after submit or the user has not submitted a response,
+    // the predict level is read only.
+    isReadonlyPredictLevel =
+      !isEditableAfterSubmit || !hasSubmittedPredictResponse;
```

Comment:
Could we make it clearer that "editable" here refers to the code and the answer to the predict question? I could see this being done by renaming the level property `editableAfterSubmit` to `codeEditableAfterSubmit`. Since that would be a pain, we could also do a similar rename to the locally scoped `isEditableAfterSubmit` on line 293 above, or even just add a comment.

Reply:
it's pretty easy to rename it at this point, I'll rename the level property!